workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -405,7 +387,7 @@ def _apply_postphysics_to_dycore_state(self) -> Diagnostics:
                 )
                 updated_state_from_tendency[TOTAL_PRECIP] = precipitation_sum(
                     self._state[TOTAL_PRECIP],
-                    diagnostics[self._postphysics_stepper.net_moistening],
```

Comment:
I just had the idea that the steppers could return the actual `net_moistening` dataarray in `__call__` so that `TimeLoop` doesn't need to know about `.stepper.label` or `.stepper.net_moistening`.

Reply:
That could work, although we would need to add a new `TimeLoop` variable to hold this quantity from the step where the tendencies are computed to where they are applied. And it feels a bit odd to add a new output to `__call__` that is already contained within one of its other outputs. My takeaway from all this is that it would be useful to enforce consistent naming of tendencies across the Stepper classes. Currently, the `PureMLStepper` outputs tendencies whose keys are `dQ1`, `dQ2`, etc. while the `Nudging` stepper outputs tendencies whose keys are `air_temperature`, `specific_humidity` etc. That would be a bigger refactor though.

Reply:
Overall with this, I am advocating that if the time-loop does a specific computation with some output of the Steppers, that it be explicitly passed that data rather than looking it up in some dict. With the former approach, the typechecker can find bugs related to moistening not being defined. Currently this would only show up as a runtime `KeyError`. 

> although we would need to add a new TimeLoop variable to hold this quantity from the step where the tendencies are computed to where they are applied.

This piece of data is already managed by the TimeLoop, so this change would only make the complexity more explicit IMO, rather than increase it overall.

>And it feels a bit odd to add a new output to __call__ that is already contained within one of its other outputs.

I agree, we would want to remove it from `diags` so that it is only in one place.